src/pudl/constants.py

```diff
@@ -1461,420 +627,19 @@
-    'MAPP',  # OLD/NEW RE Became part of MRO, resurfaced in 2010
-    'MRO',  # RE included in 2017 Form instructions
-    'NPCC',  # RE included in 2017 Form instructions
-    'RFC',  # RE included in 2017 Form instructions
-    'SERC',  # RE included in 2017 Form instructions
-    'SPP',  # RE included in 2017 Form instructions
-    'TRE',  # RE included in 2017 Form instructions (included ERCOT)
-    'WECC',  # RE included in 2017 Form instructions
-    'WSCC',  # OLD RE pre-2002 version of WECC
-    'MISO',  # ISO unclear whether technically a regional entity, but lots of entries
-    'ECAR_MAAC',
-    'MAPP_WECC',
-    'RFC_SERC',
-    'SPP_WECC',
-    'MRO_WECC',
-    'ERCOT_SPP',
-    'SPP_TRE',
-    'ERCOT_TRE',
-    'MISO_TRE',
-    'VI',  # Virgin Islands
-    'GU',  # Guam
-    'PR',  # Puerto Rico
-    'AS',  # American Samoa
-    'UNK',
-]
-
-CUSTOMER_CLASSES = [
-    "commercial",
-    "industrial",
-    "direct_connection",
-    "other",
-    "residential",
-    "total",
-    "transportation"
-]
-
-TECH_CLASSES = [
-    'backup',  # WHERE Is this used? because removed from DG table b/c not a real component
-    'chp_cogen',
-    'combustion_turbine',
-    'fuel_cell',
-    'hydro',
-    'internal_combustion',
-    'other',
-    'pv',
-    'steam',
-    'storage_pv',
-    'all_storage',  # need 'all' as prefix so as not to confuse with other storage category
-    'total',
-    'virtual_pv',
-    'wind',
-]
-
-REVENUE_CLASSES = [
-    'retail_sales',
-    'unbundled',
-    'delivery_customers',
-    'sales_for_resale',
-    'credits_or_adjustments',
-    'other',
-    'transmission',
-    'total',
-]
-
-RELIABILITY_STANDARDS = [
-    'ieee_standard',
-    'other_standard'
-]
-
-FUEL_CLASSES = [
-    'gas',
-    'oil',
-    'other',
-    'renewable',
-    'water',
-    'wind',
-    'wood',
-]
-
-RTO_CLASSES = [
-    'caiso',
-    'ercot',
-    'pjm',
-    'nyiso',
-    'spp',
-    'miso',
-    'isone',
-    'other'
-]
-
-ESTIMATED_OR_ACTUAL = {'E': 'estimated', 'A': 'actual'}
-
-TRANSIT_TYPE_DICT = {
-    'CV': 'conveyer',
-    'PL': 'pipeline',
-    'RR': 'railroad',
-    'TK': 'truck',
-    'WA': 'water',
-    'UN': 'unknown',
+    'glue': (
+        'plants_eia',
+        'plants_ferc',
+        'plants',
+        'utilities_eia',
+        'utilities_ferc',
+        'utilities',
+        'utility_plant_assn',
+    ),
 }
+"""Tables that are available in the PUDL DB, organized by data source."""
 
-# Define some categorical dtypes to reduce the size of EPA CEMS data
-epacems_codes = (
-    "LME",
-    "Measured",
-    "Measured and Substitute",
-    "Other",
-    "Substitute",
-    "Undetermined",
-    "Unknown Code",
-    "",
-)
-co2_so2_cats = pd.CategoricalDtype(
-    categories=epacems_codes,
-    ordered=False
-)
-nox_cats = pd.CategoricalDtype(
-    categories=epacems_codes + ("Calculated",),
-    ordered=False
-)
-
-column_dtypes = {
+column_dtypes: Dict[str, Dict[str, Any]] = {
```

Comment:
It should be, but I don't think it is yet. I asked about how to make this transition in some comments on big PR #806.
* One issue is that we need to compile new metadata for the EIA 861 and FERC 714 datasets, some of whose columns are represented in this dictionary already.
* Another issue is that I'm not sure there's a 1-to-1 mapping between the data types stored in the `Field` class, and the Pandas data types we want to apply to these fields.  E.g. in the case of the EPA CEMS data we're using 32-bit ints and floats since they're good enough and the dataset is huge.

Reply:
I can work on this. The 32-bit ints and floats are not a problem. These can be determined based on field constraints minimum and maximum.